sdks/java/testing/test-utils/src/main/java/org/apache/beam/sdk/testutils/NamedTestResult.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.testutils;
+
+import com.google.cloud.bigquery.LegacySQLTypeName;
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+
+/**
+ * Represents a schema and corresponding test result. Each test may have multiple named results
+ * published.
+ */
+public class NamedTestResult implements TestResult {
+
+  private final String testID;
+  private final String timestamp;
+  private final String metric;
+  private final long value;
+  private static final Map<String, String> schema =
+      ImmutableMap.<String, String>builder()
+          .put("test_id", LegacySQLTypeName.STRING.name())
+          .put("timestamp", LegacySQLTypeName.TIMESTAMP.name())
+          .put("metric", LegacySQLTypeName.STRING.name())
+          .put("value", LegacySQLTypeName.INTEGER.name())
```

Comment:
Sorry for those "scattered" comments but I was doing something similar and thought of this PR again. 

As far as I understand, the idea of `NamedTestResult` is to keep any metrics in a table with the same schema. is `INTEGER` type enough here? Maybe it's a good idea to change the type to something future-proof, like `FLOAT`? WDYT?